### PR TITLE
No longer depend on Obsolete AppInsights APIs

### PR DESCRIPTION
**Pull request by xavierdecoster**
This PR updates the Gallery (and the jobs within the solution) to use the new Application Insights initialization logic introduced in ServerCommon 2.60.0 and NuGet.Jobs (current dev branch).

Goal is to move off obsolete AppInsights APIs, such as `TelemetryConfiguration.Active`.

In doing so, some legacy code and static classes required refactoring. Work in progress, but opening the PR already to collect feedback early on.

Also needs to consume a to-be-released update of `NuGet.Services.Logging` ([PR pending](https://github.com/NuGet/ServerCommon/pull/324))

### CODE REVIEW

File: src/AccountDeleter/Job.cs [hunk 1]
```diff
@@ -127,7 +126,6 @@ protected void ConfigureGalleryServices(IServiceCollection services, IConfigurat
                 services.AddScoped<IDeleteAccountService, DeleteAccountService>();
 
                 services.AddScoped<IUserService, AccountDeleteUserService>();
-                services.AddScoped<ITelemetryClient>(sp => { return TelemetryClientWrapper.Instance; });
```
Comment (xavierdecoster) [inline src/AccountDeleter/Job.cs]: This was already configured [here](https://github.com/NuGet/NuGetGallery/pull/7745/files?utf8=%E2%9C%93&diff=unified&w=1#diff-a55e62854ff973dd124a5bf076c676edR104).

File: src/AccountDeleter/AccountDeleter.csproj [hunk 2]
```diff
@@ -107,7 +107,7 @@
       <PrivateAssets>all</PrivateAssets>
     </PackageReference>
     <PackageReference Include="NuGet.Services.Validation.Common.Job">
-      <Version>4.3.0-dev-3159955</Version>
+      <Version>4.3.0-dev-3297551</Version>
```
Comment (joelverhagen) [inline src/AccountDeleter/AccountDeleter.csproj]: Ah! I need to merge master back into dev.

File: src/NuGetGallery/App_Start/OwinStartup.cs [hunk 3]
```diff
@@ -77,63 +77,65 @@ public static void Configuration(IAppBuilder app)
 
             // Setup telemetry
             var instrumentationKey = config.Current.AppInsightsInstrumentationKey;
-            if (!string.IsNullOrEmpty(instrumentationKey))
+            var heartbeatIntervalSeconds = config.Current.AppInsightsHeartbeatIntervalSeconds;
+
+            ApplicationInsightsConfiguration applicationInsightsConfiguration;
+
+            if (heartbeatIntervalSeconds > 0)
```
Comment (skofman1) [inline src/NuGetGallery/App_Start/OwinStartup.cs]: I see that ApplicationInsightsConfiguration is initialized twice. Here and in DefaulyDependenciesModule. Why is this needed?

File: src/NuGetGallery/OData/NuGetODataController.cs [hunk 4]
```diff
@@ -27,8 +28,12 @@ public abstract class NuGetODataController
         {
             _configurationService = configurationService ?? throw new ArgumentNullException(nameof(configurationService));
             _telemetryService = telemetryService ?? throw new ArgumentNullException(nameof(telemetryService));
+
+            ODataQueryVerifier = ODataQueryVerifier.Build(telemetryService);
```
Comment (skofman1) [inline src/NuGetGallery/OData/NuGetODataController.cs]: not your change.. please pull latest dev

File: src/NuGetGallery/App_Start/DefaultDependenciesModule.cs [hunk 5]
```diff
@@ -93,30 +93,18 @@ protected override void Load(ContainerBuilder builder)
...
-                }
-                else
-                {
-                    // Configure instrumentation key,
-                    // and register NuGet.Services.Logging.TelemetryContextInitializer.
-                    ApplicationInsights.Initialize(instrumentationKey);
-                }
-            }
+            // Register the ILoggerFactory and configure AppInsights.
+            var applicationInsightsConfiguration = ConfigureApplicationInsights(configuration);
```
Comment (skofman1) [inline src/NuGetGallery/App_Start/DefaultDependenciesModule.cs]: >applicationInsightsConfiguration [](start = 16, length = 32)

Does this override the configuration we have in a applicationinsights.config? One important thing we configured is not to sample: [https://docs.microsoft.com/en-us/azure/azure-monitor/app/sampling](https://docs.microsoft.com/en-us/azure/azure-monitor/app/sampling)

Please verify that we don't get sampling back with this change. I think looking at the runtime config with debugger should be sufficient.

File: src/NuGetGallery/OData/NuGetODataController.cs (hunk 4 above)
Comment (xavierdecoster) [reply src/NuGetGallery/OData/NuGetODataController.cs]: This is my change? 😕 

`ODataQueryVerifier` used to be a static class. Not anymore. It needs a `TelemetryService` using the configured `TelemetryConfiguration`.

File: src/NuGetGallery/App_Start/DefaultDependenciesModule.cs (hunk 5 above)
Comment (xavierdecoster) [reply src/NuGetGallery/App_Start/DefaultDependenciesModule.cs]: The first thing that `ApplicationInsights.Initialize` does, is reading the `applicationinsights.config` file.

I explicitly asked about this before making the change in ServerCommon.
See the code comment [here](https://github.com/NuGet/ServerCommon/pull/321/files#diff-c0eac733d736223afbbf9061c7200bb1R59) and my question on AI issue tracker [here](https://github.com/microsoft/ApplicationInsights-dotnet/issues/1298#issuecomment-555505295).

I will verify sampling at runtime to be sure.

File: src/NuGetGallery/App_Start/OwinStartup.cs (hunk 3 above)
Comment (xavierdecoster) [reply src/NuGetGallery/App_Start/OwinStartup.cs]: I was wondering the same thing, but didn't get to digging into it more until now.

The `OwinStartup` class loads the `DefaultDependenciesModule` through a call to `app.UseAutofacInjection` [here](https://github.com/NuGet/NuGetGallery/blob/542a95582a7e306ca38fbc2e8449207e0cbb01be/src/NuGetGallery/App_Start/OwinStartup.cs#L54), which in turn calls `builder.RegisterAssemblyModules(currentAssembly);` [here](https://github.com/NuGet/NuGetGallery/blob/542a95582a7e306ca38fbc2e8449207e0cbb01be/src/NuGetGallery/App_Start/AutofacConfig.cs#L37).

My *thinking* is that we should consolidate everything within `DefaultDependenciesModule` and remove additional telemetry configuration from `OwinStartup`.

File: src/NuGetGallery/OData/NuGetODataController.cs (hunk 4 above)
Comment (skofman1) [reply src/NuGetGallery/OData/NuGetODataController.cs]: oops! Looked similar to a different change 😿
